godot-core/src/builtin/packed_array.rs

```diff
@@ -258,6 +258,34 @@ macro_rules! impl_packed_array {
                 self.as_inner().sort();
             }
 
+            /// Extracts a slice of the entire array.
+            pub fn as_slice<'a>(&'a self) -> &'a [$Element] {
+                let len = self.len();
+
+                if len == 0 {
+                    return &[];
+                }
```

Comment:
Is this case not handled by `from_raw_parts`?

Reply:
`self.ptr(0)` would panic when len is 0, because it is out of bounds.